src/pip/_internal/self_outdated_check.py

```diff
@@ -39,6 +39,15 @@ def _get_statefile_name(key: str) -> str:
     return name
 
 
+def _convert_date(isodate: str) -> datetime.datetime:
+    """Convert an ISO format string to a date.
+
+    Handles the format 2020-01-22T14:24:01Z (trailing Z)
+    which is not supported by older versions of fromisoformat.
+    """
+    return datetime.datetime.fromisoformat(isodate.replace("Z", "+00:00"))
```

Comment:
Thanks for this PR! Would it make sens to mention in the comment which Python version supports the Z suffix, so we know when we can drop this function?

Reply:
Sure. I didn't have the time to go and check, but if you want to suggest a suitable wording, I'm fine with adding it.

The reason I left things vague is that I frankly found the various discussions around the deprecation of `utcnow` (here and on the cpython lists) rather confusing, and I don't feel confident enough to make any definitive statements 🙁

Reply:
Ok, then I'm not going to be confident either :)

Reply:
Chicken! 🙂